src/loggers.jl

```diff
@@ -473,3 +474,34 @@ function Base.show(io::IO, aol::AverageObservableLogger)
             aol.current_block_size * length(aol.block_averages),
             " samples collected for observable ", aol.observable)
 end
+
+mutable struct ReplicaExchangeLogger
+    n_replicas::Int
+    n_attempts::Int
+    n_exchanges::Int
+    indices::Vector{Tuple{Int,Int}}
+    steps::Vector{Int}
+    delta::Vector{Float64}
```

Comment:
Ideally this would be `Vector{T}` with `T` as a type parameter, allowing `Float32` to be used.